File: ObjectProgrammingPython/organizm.py
class Organizm:
    MIN_ZYCIA_ABY_URODZIC = 5

    def __init__(self, Swiat, x, y, czyZyje, czyNull=True):
        self.swiat = Swiat
        self.polozenieX = x
        self.polozenieY = y
        self.czyZyje = czyZyje
        self.sila = None
        self.wiek = None
        self.inicjatywa = None
        self.oznaczenie = None
        self.czyMozeRozmnazac = None
        self.czyNull = czyNull
        self.kolor = None

    def getCzyNull(self):
        return self.czyNull

    def setCzyNull(self, czy):
        self.czyNull = czy

    def setPolozenieX(self, x):
        self.polozenieX = x

    def setPolozenieY(self, y):
        self.polozenieY = y

    def getPolozenieX(self):
        return self.polozenieX

    def getPolozenieY(self):
        return self.polozenieY

    def getInicjatywa(self):
        return self.inicjatywa

    def setInicjatywa(self, ile):
        self.inicjatywa = ile

    def getWiek(self):
        return self.wiek

    def setWiek(self, ile):
        self.wiek = ile

    def getSila(self):
        return self.sila

    def setSila(self, ile):
        self.sila = ile

    def getCzyZyje(self):
        return self.czyZyje

    def setCzyZyje(self, czy):
        self.czyZyje = czy

    def getKolor(self):
        return self.kolor

    def getOznaczenie(self):
        return self.oznaczenie

    def akcja(self):
        pass

    def kolizja(self, atakujacy):
        pass

    def przygotujOrganizm(self):
        self.wiek += 1
        self.czyMozeRozmnazac += 1

    def getCzyMozeRozmnazac(self):
        return self.czyMozeRozmnazac

    def setCzyMozeRozmnazac(self, ile):
        self.czyMozeRozmnazac = ile

    def koncowka(self):
        return " na pozycji " + str(self.polozenieX) + ":" + str(self.polozenieY) \
               + " o sile: " + str(self.sila) + " inicjatywie: " \
               + str(self.inicjatywa) + " wiek:" + str(self.wiek)

File: ObjectProgrammingJava/projekt po - 2/src/projekt/Organizm.java
package projekt;

import projekt.swiat.Swiat;

import java.awt.*;

public abstract class Organizm implements Wypisanie{
    protected int sila;
    protected int wiek;
    protected int inicjatywa;
    protected int polozenieX;
    protected int polozenieY;
    protected char oznaczenie;
    protected boolean czyZyje;
    protected int czyMozeRozmnazac;
    protected Color kolor;
    protected Swiat swiat;
    final static protected int MIN_ZYCIA_ABY_URODZIC = 5;
    public abstract String Wypisz();

    public int getSila() {
        return sila;
    }

    public void setSila(int sila) {
        this.sila = sila;
    }

    public int getWiek() {
        return wiek;
    }

    public void setWiek(int wiek) {
        this.wiek = wiek;
    }

    public int getInicjatywa() {
        return inicjatywa;
    }

    public void setInicjatywa(int inicjatywa) {
        this.inicjatywa = inicjatywa;
    }

    public int getPolozenieX() {
        return polozenieX;
    }

    public void setPolozenieX(int polozenieX) {
        this.polozenieX = polozenieX;
    }

    public int getPolozenieY() {
        return polozenieY;
    }

    public void setPolozenieY(int polozenieY) {
        this.polozenieY = polozenieY;
    }

    public char getOznaczenie() {
        return oznaczenie;
    }

    public void setOznaczenie(char oznaczenie) {
        this.oznaczenie = oznaczenie;
    }

    public boolean isCzyZyje() {
        return czyZyje;
    }

    public void setCzyZyje(boolean czyZyje) {
        this.czyZyje = czyZyje;
    }

    public int getCzyMozeRozmnazac() {
        return czyMozeRozmnazac;
    }

    public void setCzyMozeRozmnazac(int czyMozeRozmnazac) {
        this.czyMozeRozmnazac = czyMozeRozmnazac;
    }

    public boolean czyOdbilAtak(Organizm atakujacy){
        return false;
    };

    public abstract void akcja();
    public abstract void kolizja(Organizm atakujacy);

    public void przygotujOrganizm(){
        wiek++;
        czyMozeRozmnazac++;
    }

    public Color getKolor() {
        return kolor;
    }

    public void setKolor(Color color) {
        kolor = color;
    }

    protected String getKoncowka(){
        return " na pozycji " + String.valueOf(polozenieX)
                + ":" + String.valueOf(polozenieY) + " o sile: " + String.valueOf(sila)
                + " inicjatywie: " + String.valueOf(inicjatywa) + " wiek:" + String.valueOf(wiek);
    }
}
